python/cudf/cudf/core/buffer/utils.py

```diff
@@ -73,13 +97,49 @@ def as_buffer(
             "`data` is a buffer-like or array-like object"
         )
 
-    if get_option("copy_on_write"):
-        return as_exposure_tracked_buffer(data, exposed=exposed)
+    # Find the buffer types to return based on the current config
+    owner_class: Type[BufferOwner]
+    buffer_class: Type[Buffer]
     if get_global_manager() is not None:
-        return as_spillable_buffer(data, exposed=exposed)
-    if hasattr(data, "__cuda_array_interface__"):
-        return Buffer._from_device_memory(data)
-    return Buffer._from_host_memory(data)
+        owner_class = SpillableBufferOwner
+        buffer_class = SpillableBuffer
+    elif get_option("copy_on_write"):
+        owner_class = ExposureTrackedBufferOwner
+        buffer_class = ExposureTrackedBuffer
+    else:
+        owner_class = BufferOwner
+        buffer_class = Buffer
+
+    # Handle host memory,
+    if not hasattr(data, "__cuda_array_interface__"):
```

Comment:
**question:** If `data` is an `ExposureTrackedBuffer` or `SpillableBuffer`, this line will ensure that it is marked as permanently exposed. Is that what you want?

Reply:
Notice, we have the guard above:
```python
    if isinstance(data, Buffer):
        return data
```

Reply:
Ah, missed that in the diff, thanks.